### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by maccman**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import (
+  "fmt"
+  "github.com/clearbit/clearbit-go/clearbit"
+  "os"
+)
+
+func main() {
+  client := clearbit.NewClient(nil, os.Getenv("CLEARBIT_KEY"))
+
+  results, resp, _ := client.Reveal.Find(clearbit.RevealFindParams{
```
Comment (harlow) [inline README.md:19]: Don't ignore errors!! Should always look for non-nil error

File: README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+package main
+
+import (
+  "fmt"
+  "github.com/clearbit/clearbit-go/clearbit"
+  "os"
+)
+
+func main() {
+  client := clearbit.NewClient(nil, os.Getenv("CLEARBIT_KEY"))
```
Comment (harlow) [inline README.md:17]: Feels a little odd having to pass in Nil here. Could make a config struct that has sensible defaults?

**Review by lazyatom — CHANGES_REQUESTED** [review]
Where are the tests? ;-)

File: clearbit/autocomplete.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+type AutocompleteItem struct {
+	Domain string `json:"domain"`
+	Logo   string `json:"logo"`
+	Name   string `json:"name"`
+}
+
+type AutocompleteSuggestParams struct {
+	Query string `url:"query"`
+}
+
```
Comment (harlow) [inline clearbit/autocomplete.go:24]: We should add comments to anything with uppercase. That way the Godocs will generate correctly:

``` go
// AutocompleteService autocompletes a company name based on partial text.
type AutocompleteService struct {
```

File: README.md (hunk 2 above)
Comment (maccman) [reply README.md:17]: @dcadenas let's improve the config - maybe take a ClearbitConfig object that let's you set the APIKey and defaults back to pulling out the key from ENV.

**Comment by maccman** [comment]
lgtm - merge away!

**Pull request merged**